t/t6030-bisect-porcelain.sh

```diff
@@ -572,8 +572,8 @@ test_expect_success 'test bisection on bare repo - --no-checkout specified' '
 		git bisect start --no-checkout &&
 		git bisect good $HASH1 &&
 		git bisect bad $HASH4 &&
-		git bisect run eval \
-			"test \$(git rev-list BISECT_HEAD ^$HASH2 --max-count=1 | wc -l) = 0" \
+		git bisect--helper --bisect-run "test \
+		$(git rev-list BISECT_HEAD ^7b7f204a749c3125d5224ed61ea2ae1187ad046f --max-count=1 | wc -l) = 0" \
```

Comment:
I believe that this `$` would still need to be escaped with a backslash, to prevent the subshell from being evluated immediately.

But the problem is a bit bigger: as `git bisect run` is a user-facing interface, people rely on it not to be changed.

So we kind of have to retain this functionality, i.e. we need this test to be unchanged (although it is fine to change the `git bisect run` to `git bisect--helper --bisect-run` for the purposes of debugging, as you can then prefix it with `debug` to run it in GDB.

But I would like to keep the `eval` here, so that we can verify that this patch does not introduce a usage regression.

And I think we can make it work: simply by not passing `"$@"` to `--bisect-run` in the `run)` clause above, but instead `"$(git rev-parse --sq-quote "$@")"`. Can you try that?

Reply:
By escaping the `$` I was able to get the tests to pass.

though [this](https://github.com/gitgitgadget/git/pull/117/commits/62e518adb5ad293b5ca353b4fa51d30711afe2b0#diff-c98a0e600ecd8430bfe1732758ea3aa2L575) removal of `eval` persists.

> But I would like to keep the `eval` here, so that we can verify that this patch does not
> introduce a usage regression.

I agree, we need this test to be unchanged.